src/compile/mark/mark.ts

```diff
@@ -340,12 +367,154 @@ function getMarkGroup(model: UnitModel, opt: {fromPrefix: string} = {fromPrefix:
   ];
 }
 
+export function getLabel(model: UnitModel, data: string): LabelMark[] {
+  if (!model.encoding.label) {
+    return [] as LabelMark[];
+  }
+
+  const {mark} = model;
+  if (!supportLabel(mark)) {
+    log.warn(log.message.dropChannelOnMark(mark, 'label'));
+    return [] as LabelMark[];
+  }
+
+  const {label} = model.encoding;
+  const {position, avoidParentLayer, mark: _mark, method, lineAnchor, ...textEncoding} = label;
+
+  const textSpec: NormalizedUnitSpec = {
+    data: null,
+    mark: {type: 'text', ...(_mark ?? {})},
+    encoding: {text: textEncoding}
+  };
+  const textModel = new UnitModel(textSpec, null, '', undefined, model.config);
+  textModel.parse();
+
+  const {markDef, encoding, config} = textModel;
+  const clip = getFirstDefined(markDef.clip, scaleClip(model), projectionClip(model));
+  const style = getStyles(markDef);
+  const key = encoding.key;
+  const sort = getSort(model);
+  const interactive = interactiveFlag(model);
+  const aria = getMarkPropOrConfig('aria', markDef, config);
+
+  return [
+    {
+      name: model.getName('marks_label'),
+      type: markCompiler.text.vgMark as 'text',
+      ...(clip ? {clip: true} : {}),
+      ...(style ? {style} : {}),
+      ...(key ? {key: key.field} : {}),
+      ...(sort ? {sort} : {}),
+      ...(interactive ? interactive : {}),
+      ...(aria === false ? {aria} : {}),
+      from: {data},
+      encode: {
+        update: {
+          ...(omit(
+            encodeBaseEncodeEntry(textModel, {
+              align: 'ignore',
+              baseline: 'ignore',
+              color: 'include',
+              size: 'ignore',
+              orient: 'ignore',
+              theta: 'ignore'
+            }),
+            ['x', 'y', 'angle', 'radius', 'theta']
+          ) as TextEncodeEntry),
```

Comment:
Why do we need to cast to `as TextEncodeEntry`? Can we avoid casting?

Same for below.

Reply:
I defined the return type to be `LabelMark` which is a `TextMark` with a label-transform.
I think we could avoid this cast by loosen the definition of `LabelMark` to be just a mark with a label-transform.